modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EBinary.java

```diff
@@ -280,6 +285,35 @@ private void analyzeSub(Locals variables) {
         actual = promote;
     }
 
+    private void analyzeFind(Locals variables) {
+        analyzeRegexOp(variables);
+        /*
+         * Constant detection isn't really a thing for FIND because it returns a mutable Matcher. Unless we can detect that it is being
```

Comment:
I'm pretty sure it isn't the right phrase for it. That stuff painless does where it maintains the `.constant` member. Like in `analyzeSup` above.